### DISCUSSION TITLE
Pulseaudio Failed to load module "module-native-protocol-unix"

### DISCUSSION THREAD

**Discussion by Sniptzler** [dtcooper/raspotify#649] [Q&A]
I'm currently trying to configure raspotify with pulseaudio on a freshly installed Raspberry Pi. I can't figure out why pulseaudio can't load this module which is neccessary for Raspotify to work. I followed the instructions provided by the wiki. Hopefully this is not a stupid question. Thanks for any help.

**Comment by Sniptzler** [comment]
after checking with ```systemctl status pulseaudio```
```
Jan 01 19:39:18 rasp-desk pulseaudio[580]: W: [pulseaudio] authkey.c: Failed to open cookie file '/var/run/pulse/.config/pulse/cookie': No such file or directory
Jan 01 19:39:18 rasp-desk pulseaudio[580]: W: [pulseaudio] authkey.c: Failed to load authentication key '/var/run/pulse/.config/pulse/cookie': No such file or directory
Jan 01 19:39:18 rasp-desk pulseaudio[580]: W: [pulseaudio] authkey.c: Failed to open cookie file '/var/run/pulse/.pulse-cookie': No such file or directory
Jan 01 19:39:18 rasp-desk pulseaudio[580]: W: [pulseaudio] authkey.c: Failed to load authentication key '/var/run/pulse/.pulse-cookie': No such file or directory
Jan 01 19:39:18 rasp-desk pulseaudio[580]: E: [pulseaudio] socket-server.c: bind(): Address already in use
Jan 01 19:39:18 rasp-desk pulseaudio[580]: E: [pulseaudio] module.c: Failed to load module "module-native-protocol-unix" (argument: "auth-cookie-enabled=0 auth-anonymous=1"): initialization failed.
Jan 01 19:40:47 rasp-desk pulseaudio[580]: W: [pulseaudio] protocol-native.c: Denied access to client with invalid authentication data.
```

**Reply by overfl0** [reply]
I know you posted your problem two years ago, but since this is where you end up when googling for that error, here we go:
Your problem is that you pasted your `load-module module-native-protocol-unix auth-cookie-enabled=0 auth-anonymous=1` line at the end of the `/etc/pulse/system.pa` file, while there **already** is a `load-module module-native-protocol-unix` in that file.

As a result, pulseaudio loads the module twice, tries to bind to `/var/run/pulse/native` twice and fails during the second try. That's why you get the "Address already in use" error.

The solution is to just remove the first `load-module` line.

**Comment by maxpain** [comment]
The same problem

**Comment by RayMoonFox** [comment]
same problem

**Reply by RayMoonFox** [reply]
not the authentication key part, just the error
